trainings/sourcegraph-101/talk-track.md

```diff
@@ -0,0 +1,327 @@
+
+*Add quotes, rerun the search.*
+
+So if I take them back out, you can see that what I'm seeing is exact matches already.
+
+*Take them back out, rerun it again.*
+
+### The search mode buttons
+
+In addition to our default literal, or exact match, search mode, Sourcegraph offers two other search modes: regular expression, and structural search. To run a search in either of these modes, you’ll use the appropriate icons in the search bar, on the right-hand side of the bar, next to the search button. 
+
+To run your search using [RE2 regex syntax](https://github.com/google/re2/wiki/Syntax), click the .* icon before running your search. If you click that icon after running your search, your search will re-run as a regex search. In regular expression searches, space characters are handled as fuzzy matches. So you see here that if I run that `new auth provider` search as a regex, I'm now getting matches for `new authz provider`, `newOAuthProvider`, etc. I can also enforce word boundaries, so if I run a search for `new \bauth\b provider`, I'm only going to see results where auth is not a substring.
+
+*Search for context:global new \bauth\b provider patternType:regexp , show the results*
+
+The final search mode option is our language-aware structural search option, represented by the square bracket button ([]). When selected, you’ll have the option to include structural holes, represented by … or :[a] in your query, to run multi-line fuzzy searches. 
+
+*Search for context:global lang:JavaScript class ... extends ... {...} count:all patternType:structural*
+
+So, for example, if I search for `context:global lang:JavaScript class ... extends ... {...} count:all patternType:structural`, I get a really wide variety of results that match that pattern. Something to call out is that structural search mode is smart enough to match nested characters—so you can see it won't prematurely close on the first closing curly brace it sees; instead, it will wait until it finds the matching one.
+
+I'm going to pause here: are there any questions about our three search modes, or do folks have use cases that jump to mind for any of them?
+
+*Pause, leave space for questions, get a drink of water*
+
+### The search results page
+
+Awesome! So, let's chat through what you'll see when you run the search. When you run your search, you will be redirected to a search results page. It will look something like this:
+
+*run the `new auth provider` regex search again*
+
+The main area of the page will show your search results, with the matching text highlighted. By default, you’ll see one line above and below your matching line, in order to provide some context on the search result. The number of lines shown can be customized in your user settings page, if you prefer. Above the search result, you’ll see the repo name and the file path within the repo. If your match is for a file path or repo name, you’ll see just a link to that repo or file, with no preview code.
```

Comment:
Yeah, I think it's just unclear without a visual aid. If you read the line looking at the search results, does it make sense?

Reply:
Yeah, I think with a visual aid it's perfectly fine. We'll just have to keep that in mind whenever this is made into a LMS type of course or if we ever have the course material available later that adding some images could be useful here.

To note, I'm a fan of being careful how many images are added because it's technical debt if UI changes occur, haha.